modules/psmisc/23.7/overlay/defs.bzl

```diff
@@ -0,0 +1,36 @@
+load("@bazel_tools//tools/build_defs/cc:action_names.bzl", "ACTION_NAMES")
+load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
+
+def _make_signames_impl(ctx):
+    cc_toolchain = find_cpp_toolchain(ctx)
+    conf = cc_common.configure_features(ctx = ctx, cc_toolchain = cc_toolchain)
+    cpp = cc_common.get_tool_for_action(feature_configuration = conf, action_name = ACTION_NAMES.c_compile)
+    signames_c = ctx.actions.declare_file("signames.c")
+    ctx.actions.write(signames_c, "#include <signal.h>")
+    script = ctx.actions.declare_file("gen_signames_h.sh")
+    ctx.actions.write(script, """\
+#!/bin/sh
+"$1" -dM -E "$2" | 
```

Comment:
This looks pretty magical, could you add a comment explaining where it comes from? Why is a c compiler needed here?

This could also pose issues with cross compilation. Could we pregenerate the list and check it in instead?

Reply:
ah yes this really needs a comment. It's from https://gitlab.com/psmisc/psmisc/-/blob/v23.7/Makefile.am?ref_type=tags#L104-112

Reply:
I'm not sure checking in is the best choice because I don't know if available signals might differ between OSes

Reply:
Basically what's happening here is that it uses the preprocessor to extract all the signal defines from the system's signal.h and then generates a C array with their names.

Reply:
Since it does run with a c++ toolchain selected for the target platform, this actually likely works with cross-platform builds.